internal/decision/engine.go

```diff
@@ -201,6 +203,10 @@ type TaskInfo struct {
 // It should return true if task 'ta' has higher priority than task 'tb'
 type TaskComparator func(ta, tb *TaskInfo) bool
 
+// PeerBlockRequestFilter is used to accept / deny requests for a CID coming from a PeerID
+// It should return true if the request should be fullfilled.
+type PeerBlockRequestFilter func(p peer.ID, c cid.Cid) bool
```

Comment:
Are we fine using values here? My intuition is that it's safer to move around values rather than pointers.

Reply:
That function signature looks good to me.